crates/api_server/src/server.rs

```diff
@@ -105,34 +87,18 @@ impl NodeServerBuilder {
             .set_rpc_middleware(RpcServiceBuilder::new().rpc_logger(100));
 
         let server = server_builder.build(addr).await.unwrap();
-        let rpc = self.rpc.clone();
-        self.server_futs
-            .push(Box::pin(async move { server.start(rpc) }));
-    }
-
-    pub async fn run(self) -> NodeServerHandle {
-        let handles = futures::future::join_all(self.server_futs).await;
-        NodeServerHandle { handles }
+        let rpc = Self::default_rpc(self.node);
+        NodeServer(Box::new(move || server.start(rpc)))
     }
 }
 
-/// Node's server handle.
-///
-/// When all [`NodeServerHandle`]'s have been `dropped` or `stop` has been called
-/// the server will be stopped.
-#[derive(Debug, Clone)]
-pub struct NodeServerHandle {
-    handles: Vec<ServerHandle>,
-}
-
-impl NodeServerHandle {
-    /// Tell the server to stop without waiting for the server to stop.
-    pub fn stop(&self) -> Result<(), AlreadyStoppedError> {
-        self.handles.iter().try_for_each(|handle| handle.stop())
-    }
+pub struct NodeServer(Box<dyn FnOnce() -> ServerHandle>);
 
-    /// Wait for the server to stop.
-    pub async fn stopped(self) {
-        futures::future::join_all(self.handles.into_iter().map(|handle| handle.stopped())).await;
+impl NodeServer {
+    /// Start responding to connections requests.
+    ///
+    /// This will run on the tokio runtime until the server is stopped or the `ServerHandle` is dropped.
```

Comment:
```suggestion
    /// This will run on the tokio runtime until the server is stopped or the `ServerHandle` is dropped.
    ///
    /// See [`ServerHandle`](https://docs.rs/jsonrpsee-server/0.24.7/jsonrpsee_server/struct.ServerHandle.html) docs for more details.
```